src/services/users.service.js

```diff
@@ -0,0 +1,41 @@
+const { getNumberId } = require('./../utils');
+
+class Users {
+  constructor(initialUsers) {
+    this._users = initialUsers || [];
+  }
+
+  getAll() {
+    return this._users;
+  }
+
+  getById(id) {
+    return this._users.find((user) => user.id === id) || null;
+  }
+
+  create(name) {
+    const newUser = { name, id: getNumberId() };
+
+    this._users.push(newUser);
+
+    return newUser;
+  }
+
+  update(id, data) {
+    const userToUpdate = this.getById(id);
+
+    Object.assign(userToUpdate, { ...data });
```

Comment:
In the `update` method, consider checking if `userToUpdate` is null before attempting to update it. This will prevent errors if the user with the given ID does not exist.